src/detail-page/components/TabJobContent/JobTab.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { Button, CircularProgress, Input } from '@material-ui/core';
+import JobListItem from './JobListItem';
+import useFetchJobs from './useFetchJobs';
+import Job from '../../../data/Jobs';
+import hiringPicture from '../../../icons/hiring.svg';
+
+const SearchInput = styled(Input)`
+  width: 250px;
+  height: 35px;
+  border: 1px #CFCFCF solid;
+  margin: ${props => props.theme.spacing(5)}px 0 ${props => props.theme.spacing(1.5)}px ${props => props.theme.spacing(1.5)}px ;
+  padding-left: 0 ${props => props.theme.spacing(1.5)}px 
+`;
+
+const SearchButton = styled(Button)`
+  width: 100px;
+  margin: ${props => props.theme.spacing(1.5)}px;
+  display: block;
+  color: white;
+`;
+
+const HiringImage = styled('object')`
+  width: 250px;
+  float: right;
+  margin: 0 ${props => props.theme.spacing(10)}px ${props => props.theme.spacing(10)}px 0;
+`;
+
+const SearchSpinner = styled('div')`
+  margin: ${props => props.theme.spacing(5)}px auto;
+  width: 50%;
+`;
+
+const JobTab: React.FC = () => {
+  const [role, setRole] = React.useState('');
+  const [city, setCity] = React.useState('');
+  const URL = `https://cors.bridged.cc/https://jobs.github.com/positions.json?title=${role}&location=${city}`;
+  const { jobPosts, isFetching } = useFetchJobs(URL);
+  const handleChangeRole = React.useCallback(e => setRole(e.target.value), []);
+  const handleChangeCity = React.useCallback(e => setCity(e.target.value), []);
+
+  return (
+    <>
+      <SearchInput type="search" placeholder="Job title" value={role} disableUnderline onChange={handleChangeRole} />
+      <SearchInput placeholder="City" disableUnderline value={city} onChange={handleChangeCity} />
+      <SearchButton color="secondary" variant="contained" onClick={useFetchJobs(URL)}>Search</SearchButton>
```

Comment:
Unfortunately, we can't call a React hook from here, because we'd be breaking [Rules of Hooks](https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level).

We need a bit of mental gymnastics here and think about it the hooks way. Specifically: we can't call a hook directly via onClick. Hooks fire on changes to hooks deps array, in our case this array includes `url` value:

![image](https://user-images.githubusercontent.com/8350925/114370761-1bd51280-9b80-11eb-9642-0397264a8079.png)

If the hook will fire on change to url, all we have to do is update url on click. Instead of hardcoding url as a string, let's set it via `setState` and update state on click:
```
const [url, setUrl] = useState('');
...
<SearchButton color="secondary" variant="contained" onClick={() => setUrl(`https://cors.bridged.cc/https://jobs.github.com/positions.json?title=${role}&location=${city}`)}>Search</SearchButton>

Reply:
I tried to do it this way, but it doesn't seem to work. Link changes after clicking, but the api stuff is not being uploaded to jobPosts. The link also only changes when I double-click.

Reply:
done;)